File: virtual-world-cpp/VirtualWorld/Plants/Dandelion.cpp
#include "Dandelion.h"

Dandelion::Dandelion(int x, int y, World *world) : Plant(x, y, 0, 0, dandelionCode, world) {}

Plant *Dandelion::Clone(int x, int y, World *world) {
    return new Dandelion(x, y, world);
}

std::string Dandelion::GetName() {
    return "Dandelion";
}

void Dandelion::Action() {
    for(int i = 0; i < 3; i++)
        Plant::Action();
}

File: virtual-world-java/src/main/java/me/knasuta/virtualworld/simulation/plants/Dandelion.java
package me.knasuta.virtualworld.simulation.plants;

import me.knasuta.virtualworld.simulation.Organism;
import me.knasuta.virtualworld.simulation.Point;
import me.knasuta.virtualworld.simulation.World;

import java.awt.*;

public class Dandelion extends Plant {
    public Dandelion(Point location, World world) {
        super(location, 0, 0, world);
    }
    @Override
    public String getName() {
        return "Dandelion";
    }

    @Override
    public Organism Clone(Point location, World world) {
        return new Dandelion(location, world);
    }

    @Override
    public Color getColor() {
        return Color.YELLOW;
    }

    @Override
    public void Action(){
        for(int i = 0; i < 3; i++)
            super.Action();
    }
}
